### PR TITLE
Alleles + Strains widgets for Laboratory class.

### PR DESCRIPTION
**Pull request by azurebrd**

### CODE REVIEW

File: src/rest_api/classes/laboratory/widgets/alleles.clj [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        data (->> (d/q '[:find [?varent ...]
+                         :in $ ?laboratory
+                         :where 
+                         [?laboratory :laboratory/alleles ?varent] ]
+                       db (:db/id laboratory))
+                  (map (fn [wbvar]
+                         (let [allele (d/entity db wbvar)]
+                           (pace-utils/vmap
+                             :sequenced 
+                             (if (= (str (:variation/seqstatus allele))
```
Comment (mgrbyte) [inline src/rest_api/classes/laboratory/widgets/alleles.clj]: No need to cast to `str` here for equality testing, can simply do:
```clojure
(if (= (:variation/seqstatus allele) :variation.seqstatus/sequenced)
   "yes"
   "no")
```

File: src/rest_api/classes/laboratory/widgets/alleles.clj [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                                    ":variation.seqstatus/sequenced") "yes" "no")
+                             :type 
+                             (if (:method/id (:locatable/method allele))
+                               (-> (:method/id (:locatable/method allele))
+                                   (str/replace #"_allele" "")
+                                   (str/replace #"Allele" "")))
```
Comment (mgrbyte) [inline src/rest_api/classes/laboratory/widgets/alleles.clj]: No problem here, but could use `some->` to be more succinct:
```clojure
:type (some-> (:method/id (:locatable/method allele))
              (str/replace #"_allele" "")
              (str/replace #"Allele" ""))
```

**Review by mgrbyte — CHANGES_REQUESTED** [review]
Generally looks good.
One change and one suggestion :+1:

**Review by adamjohnwright — APPROVED** [review]

**Comment by adamjohnwright** [comment]
I agree with Matts suggestions. I don't see anything else, in that way of style, that needs to change. Great work

File: src/rest_api/classes/laboratory/widgets/alleles.clj (hunk 1 above)
Comment (azurebrd) [reply src/rest_api/classes/laboratory/widgets/alleles.clj]: Ah, cool, thanks.

File: src/rest_api/classes/laboratory/widgets/alleles.clj (hunk 2 above)
Comment (azurebrd) [reply src/rest_api/classes/laboratory/widgets/alleles.clj]: Oooh, that's really neat, thanks Matt !  I've made the changes, I appreciate the feedback while travelling =)

File: src/rest_api/classes/laboratory/widgets/alleles.clj [hunk 3]
```diff
@@ -16,13 +16,11 @@
                          (let [allele (d/entity db wbvar)]
                            (pace-utils/vmap
                              :sequenced 
-                             (if (= (str (:variation/seqstatus allele))
+                             (if (= (:variation/seqstatus allele)
                                     ":variation.seqstatus/sequenced") "yes" "no")
```
Comment (mgrbyte) [inline src/rest_api/classes/laboratory/widgets/alleles.clj]: Value being compared shouldn't be a string (is it?) (i.e: ":variation.seqstatus/sequenced" _should_ be a keyword/ident not a string)

**Comment by azurebrd** [comment]
@mgrbyte Right !  Thanks for catching that, it seemed odd that it didn't need a (str) casting earlier, and I didn't realize your suggestion was comparing it straight to a keyword/ident.  Thanks !

**Review by mgrbyte — APPROVED** [review]

**Pull request merged**